python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -130,21 +140,25 @@ def _calculate(
         """
         # Convert centroid to GalSim's PositionD type.
         guessCentroid = galsim.PositionD(centroid.x, centroid.y)
-
         try:
             # Attempt to compute HSM moments.
-            shape = galsim.hsm.FindAdaptiveMom(
+
+            # Use galsim c++/python interface directly.
+            shape = galsim.hsm.ShapeData()
+            hsmparams = galsim.hsm.HSMParams.default
+
+            galsim._galsim.FindAdaptiveMomView(
+                shape._data,
                 image,
-                weight=weight,
-                badpix=badpix,
-                guess_sig=sigma,
-                precision=precision,
-                guess_centroid=guessCentroid,
-                strict=True,
-                round_moments=self.config.roundMoments,
-                hsmparams=None,
+                weight_image,
+                float(sigma),
+                float(precision),
+                guessCentroid._p,
+                bool(self.config.roundMoments),
```

Comment:
Are casting necessary here, especially this one?

Reply:
The casting is just-in-case to make sure that we get the precise datatype match to what the c++ expects.  In these cases I always cast to avoid any ... unpleasant surprises.